### PR TITLE
contrib: refactor fetching of wallet token balances using Redux

### PR DESCRIPTION
**Pull request by 0xcryptovenom**
- add `./src/selectors` directory for Redux selectors
- add `selectWalletPublicKey` memoized selector (using `reselect` through `@reduxjs/toolkit`)`
- simplify fetching of wallet token balances 
- remove `useWatchWalletBalance` hook in favor of a `fetchWalletTokenBalances` thunk action.
  - remove props drilling from the App component to all component which may need to fetch token balances by using the `fetchWalletTokenBalances` thunk action.
    - the same pattern can be applied to most custom hooks in the code base.
      - it allows separating business logic from the UI, better architecture which also improves testabibility
      - it will have a good impact on rendering performance with wider adoption in the code base.
    - clean-up "walletBalances" actions:
      - use type inference
      - define constant action type & use it in the matching reducer
      - use concise definition of action creators: a simple function returning an action object
      - follow-up of https://github.com/AdrenaFoundation/frontend/pull/396
- fix token balances should be reset immediately when the user disconnects.

### CODE REVIEW

File: src/actions/thunks.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
+import type { ThunkAction } from '@reduxjs/toolkit';
+import { NATIVE_MINT } from '@solana/spl-token';
+import BN from 'bn.js';
```
Comment (0xcryptovenom) [inline src/actions/thunks.ts:3]: `import { BN } from '@coral-xyz/anchor';`

File: src/selectors/wallet.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { PublicKey } from '@solana/web3.js';
+import { createSelector } from 'reselect';
```
Comment (0xcryptovenom) [inline src/selectors/wallet.ts:2]: `import { createSelector } from '@reduxjs/toolkit';`

**Comment by 0xcryptovenom** [comment]
Latest commit fixes the typechecking issue, preview build should be good now.

**Comment by 0xcryptovenom** [comment]
@Adrena-Corto @adrena-orex sorry folks I had forgotten to make the fetching of wallet token balances reactive to user wallet change, should be good now - though tbh I haven't tested cuz of https://github.com/AdrenaFoundation/frontend/issues/399

File: src/actions/thunks.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+import { selectWalletPublicKey } from '@/selectors/wallet';
+import type { Dispatch, RootState } from '@/store/store';
+import type { TokenSymbol } from '@/types';
+import { findATAAddressSync, nativeToUi } from '@/utils';
+
+import { setWalletTokenBalances } from './walletBalances';
+
+export const fetchWalletTokenBalances =
+  () => async (dispatch: Dispatch, getState: () => RootState) => {
+    const connection = window.adrena.client.connection;
```
Comment (0xcryptovenom) [inline src/actions/thunks.ts:14]: Looks like for some reason, the connection is still `null` when this runs, even after connecting the wallet.

Looking into it.

Comment (0xcryptovenom) [reply src/actions/thunks.ts]: Should be fixed now, using the global main connection instead, which is set on app initialization, way earlier.

Comment (0xcryptovenom) [reply src/actions/thunks.ts]: Fixed, tested locally, now behaves as expected.

**Comment by Adrena-Corto** [comment]
Got an issue where I cannot see my token balance anymore (so cannot open a position)

**Pull request merged**